### ISSUE TITLE
Cannot draw molecule which includes an atom with a `[!#X]` query (for any X)

### ISSUE THREAD

**Issue by ShangChien** [rdkit/rdkit#6033]
**Describe the bug**
`m = Chem.MolFromSmiles('CO*.C1=CC=NC=C1 |c:2,4,6,m:2:3.5.4|')` can work well, but execute `m`  with error:
![image](https://user-images.githubusercontent.com/48350636/215665887-b22534d0-959e-4f05-b367-14bc8704730e.png)
full error :
```
[12:36:57] 

****
Pre-condition Violation
bad query type
Violation occurred on line 1169 in file C:\rdkit\build\temp.win-amd64-cpython-38\Release\rdkit\Code\GraphMol\MolDraw2D\DrawMol.cpp
Failed Expression: atom.getQuery()->getDescription() == "AtomOr"
****

Output exceeds the [size limit](command:workbench.action.openSettings?[). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?64f7c3ce-2006-40ac-a17b-79e30aced92d)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File d:\conda\lib\site-packages\IPython\core\formatters.py:343, in BaseFormatter.__call__(self, obj)
    341     method = get_real_method(obj, self.print_method)
    342     if method is not None:
--> 343         return method()
    344     return None
    345 else:

File d:\conda\lib\site-packages\rdkit\Chem\Draw\IPythonConsole.py:155, in _toPNG(mol)
    153   highlightAtoms = []
    154 kekulize = kekulizeStructures
--> 155 return Draw._moltoimg(mol, molSize, highlightAtoms, "", returnPNG=True, kekulize=kekulize,
    156                       drawOptions=drawOptions)

File d:\conda\lib\site-packages\rdkit\Chem\Draw\__init__.py:489, in _moltoimg(mol, sz, highlights, legend, returnPNG, drawOptions, **kwargs)
    486   d2d.DrawMolecule(mc, legend=legend or "", highlightAtoms=highlights or [],
    487                    highlightBonds=bondHighlights)
    488 else:
--> 489   d2d.DrawMolecule(mc, legend=legend or "", highlightAtoms=highlights or [])
    490 d2d.FinishDrawing()
    491 if returnPNG:
RuntimeError: Pre-condition Violation
	bad query type
...
	Violation occurred on line 1169 in file Code\GraphMol\MolDraw2D\DrawMol.cpp
	Failed Expression: atom.getQuery()->getDescription() == "AtomOr"
	RDKIT: 2022.09.4
	BOOST: 1_78
<rdkit.Chem.rdchem.Mol at 0x27c4e1fe7b0>
```
**Expected behavior**
![image](https://user-images.githubusercontent.com/48350636/215666878-3c6b2829-cad2-4fcc-9b59-947cd274909d.png)

Version：rdkit-2022.9.4

**Comment by ShangChien** [comment]
#5980 related issue

**Comment by greglandrum** [comment]
> #5980 related issue

That issue is not related to this one

**Comment by greglandrum** [comment]
Confirmed. we will look into this

**Comment by ShangChien** [comment]
For poor my code review skill, i can only provide user's info: For above whole enumrate mols process, rdkit-2022.3.* work very well. 2022.9.* may not stable for it.

**Comment by greglandrum** [comment]
Here's a shorter reproducible:
```
In [23]: qm = Chem.MolFromSmarts('[!#1]')

In [24]: d2d = Draw.MolDraw2DSVG(-1,-1)

In [25]: d2d.DrawMolecule(qm)
[06:35:19]

****
Pre-condition Violation
bad query type
Violation occurred on line 1169 in file D:\bld\rdkit_1674051513102\work\Code\GraphMol\MolDraw2D\DrawMol.cpp
Failed Expression: atom.getQuery()->getDescription() == "AtomOr"
****

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In [25], line 1
----> 1 d2d.DrawMolecule(qm)

RuntimeError: Pre-condition Violation
        bad query type
        Violation occurred on line 1169 in file Code\GraphMol\MolDraw2D\DrawMol.cpp
        Failed Expression: atom.getQuery()->getDescription() == "AtomOr"
        RDKIT: 2022.09.4
        BOOST: 1_78
```

**Comment by greglandrum** [comment]
This was fixed by #6055
